lib/roby/app.rb

```diff
@@ -1041,6 +1047,44 @@ def base_setup
             @base_setup_done = true
         end
 
+        def lock_log_dir
+            return if log_dir_locked?
+
+            return unless @lock_file
+
+            @lock_file ||= File.open(
+                File.join(@log_dir, LOCK_FILE_EXT),
+                File::RDWR | File::CREAT, 0o644
+            )
+            @lock_file.flock(File::LOCK_EX | File::LOCK_NB)
+        rescue Errno::ENOENT, Errno::EACCES, RuntimeError => e
+            warn "Failed to lock directory #{@log_dir}: #{e.message}"
+        end
+
+        def unlock_log_dir
+            return unless log_dir_locked?
+
+            return unless @lock_file
+
+            @lock_file.flock(File::LOCK_UN)
+            @lock_file.close
+            @lock_file = nil
+        rescue Errno::ENOENT, Errno::EACCES, RuntimeError => e
+            warn "Failed to unlock directory #{@log_dir}: " \
+                 "#{e.message}"
+        end
+
+        def log_dir_locked?
+            return unless @lock_file
+
+            File.open(@lock_file) do |file|
+                return !file.flock(File::LOCK_EX | File::LOCK_NB)
+            end
+        rescue Errno::ENOENT, Errno::EACCES => e
+            warn "Failed to check lock status for log directory " \
+                 "#{@log_dir}: #{e.message}"
```

Comment:
Right ... but ... what are you returning in this case ? How should the caller *react* to this state of things ?

This should be a hard error. Just let the exception flow, and make sure the method is called at the right place.